File: Abstraction/Models/FilterSearchModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Abstraction.Models
{
    public class FilterSearchModel
    {
        public int? CategoryId { get; set; }

        public int? MinPrice { get; set; }

        public int? MaxPrice { get; set; }
    }
}

File: angular/src/app/shared/models/filter-search.model.ts
export class FilterSearchModel {
  categoryId?: number | null;
  minPrice?: number | null;
  maxPrice?: number | null;

  constructor(
    categoryId?: number | null,
    minPrice?: number | null,
    maxPrice?: number | null
  ) {
    this.categoryId = categoryId;
    this.minPrice = minPrice;
    this.maxPrice = maxPrice;
  }
}
